webpack/assets/javascripts/react_app/components/factCharts/index.js

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Modal, Button, OverlayTrigger, Tooltip } from 'patternfly-react';
+import PropTypes from 'prop-types';
+import helpers from '../../common/helpers';
+import DonutChart from '../common/charts/DonutChart';
+import Loader from '../common/Loader';
+import MessageBox from '../common/MessageBox';
+import { STATUS } from '../../constants';
+import * as FactChartActions from '../../redux/actions/factCharts/';
+
+import { navigateToSearch } from '../../../services/ChartService';
+
+class FactChart extends React.Component {
+  constructor(props) {
+    super(props);
+    helpers.bindMethods(this, ['openModal', 'closeModal']);
+  }
+
+  openModal() {
+    const { showModal, getChartData, data: { id, path, title } } = this.props;
+
+    getChartData(path, id);
+    showModal(id, title);
+  }
+
+  closeModal() {
+    this.props.closeModal();
+  }
+
+  render() {
+    const {
+      factChart,
+      data: { id, title, search },
+      modalToDisplay,
+    } = this.props;
+
+    const handleChartClick =
+      search && search.match(/=$/) ? null : navigateToSearch.bind(null, search);
+
+    const chartProps = {
+      data: factChart.chartData ? factChart.chartData : null,
+      key: `chart-${id}`,
+      onclick: handleChartClick,
+    };
+
+    const chart = <DonutChart {...chartProps} config='large' />;
```

Comment:
if the field has a long name, it's not readable, I'm not sure if its the c3 impementation (e.g. we need to open an issue on patternfly-react or somewhere else) but this its how it looks:
![image](https://user-images.githubusercontent.com/20167/37874288-3b557bfe-3035-11e8-87af-0fd3d86540d8.png)

vs

![image](https://user-images.githubusercontent.com/20167/37874290-47d329bc-3035-11e8-8e6a-44cd1883256b.png)

Reply:
I and @amirfefer think we should open an issue on that